contrib/cirrus/runner.sh

```diff
@@ -375,7 +375,7 @@ function _bail_if_test_can_be_skipped() {
 
     # Cirrus sets these for PRs but not branches or cron. In cron and branches,
     #we never want to skip.
-    for v in CIRRUS_CHANGE_IN_REPO CIRRUS_PR; do
+    for v in CIRRUS_CHANGE_IN_REPO CIRRUS_PR DEST_BRANCH; do
```

Comment:
Good to have this here, [but `lib.sh` will default the value to `main`.](https://github.com/containers/podman/blob/main/contrib/cirrus/lib.sh#L77)  This was originally done for the benefit of `hach/get_ci_vm.sh` users, however the current/newer version of that script will automatically grab the value from `.cirrus.yml`.  So I think the defaulting definition in `lib.sh` can simply be removed.  However it might be good to add a safety-net: [a non-empty requirement to `setup_environment.sh`.](https://github.com/containers/podman/blob/main/contrib/cirrus/setup_environment.sh#L27-L28) just to be sure.

Reply:
> Can we guarantee that every CI environment sets that?

Probably we can, that's why I suggested adding it to `req_env_var` - as a guarantee it.  Every CI task and script which calls `runner.sh` always first calls `setup_environment.sh` (including a user running `hack/get_ci_vm.sh`).  So that would be the best fool-proofing we could do here - (for example) in case someone removes `$DEST_BRANCH` from `.cirrus.yml`.  The "is it empty" check in `_bail_if_test_can_be_skipped()` will be a good second-line of defense, we should keep that too.

Reply:
So, actually, we can't (guarantee). My previous run failed in the `f35/f36 container` test, because `req_env_var` didn't find `DEST_BRANCH`. So I removed it and we'll see what happens here.

Reply:
Oh right, because `setup_environment.sh` isn't saving it into `/etc/ci_environment` for re-use.  Right, this is what I discovered in #14131, now I remember.  That's why it was only affecting rootless and isn't showing up in containers.  Both of those load their env. var.s from `/etc/ci_environment`.  So we have to set it in there or it will be lost.

Reply:
> So we have to set it in there or it will be lost.

I can't tell if this means "Ed needs to do something" or "Chris needs to do something." After looking at the `contrib/cirrus` scripts and getting pretty lost, I'm settling on "Chris needs to do something".

Reply:
https://github.com/containers/podman/pull/14179